$NetBSD$

This is required to enable builds in architectures other than amd64
(x86_64). Currently supports i386/armv6/armv7.

The upstream as of now "hardcodes" amd64 (x86_64) as the only build
option.

Now it does a host CPU check and picks the architecture accordingly.

XXX: Should we add arm / aarch64? Since the library has the means to
support it?

XXX: What is the difference between arm_32 and arm_neon?

XXX: Is arm cpu detection correct?

--- configure.ac.orig	2019-03-11 19:42:59.000000000 +0000
+++ configure.ac
@@ -12,6 +12,22 @@ AM_INIT_AUTOMAKE([-Wall -Werror foreign 
 AM_PROG_AR
 LT_INIT
 
+dnl Check the machine architecture.
+AC_CANONICAL_HOST
+
+AS_CASE([$host_cpu],
+  [i?86], [ARCH_NAME=arch_32],
+  [x86_64], [ARCH_NAME=arch_x86_64],
+  [amd64], [ARCH_NAME=arch_x86_64],
+  [armv6], [ARCH_NAME=arch_arm_32],
+  [armv7], [ARCH_NAME=arch_arm_32],
+  [AC_MSG_ERROR("No assembler code for CPU $host_cpu")]
+)
+
+AC_SUBST([ARCH_NAME])
+
+AM_CONDITIONAL([ARCH_X86], [test "x$ARCH_NAME" = xarch_x86_64] || [test "x$ARCH_NAME" = xarch_32])
+
 dnl Checks for programs.
 AC_PROG_CXX
 AC_PROG_CC
